lms/static/completion/js/ViewedEvent.js

```diff
@@ -0,0 +1,180 @@
+/** Ensure that a function is only run once every `wait` milliseconds */
+function throttle(fn, wait) {
+  let time = 0;
+  function delay() {
+    const now = Date.now();
+    if (time + wait < now) {
```

Comment:
Can you explain why `time+wait < now` then we execute the wrapped function? I guess I don't understand the logic. Shouldn't it be `time+wait > now`?

Reply:
Hah, that tripped me up the first time, too, but it actually makes sense as written.  Before the function is called the first time, `time+wait` will equal `wait`, which is definitely less that `now`.  So the function is called the first time without any delay.  Now we set `time = now`.  So now we'll have to wait for `wait` ms in order for `now` to be greater than `time + wait`.

Reply:
Adding an explanatory comment.

Reply:
I see. Thank you for the explanation @iloveagent57. I didn't know that this function is intended to be called at least twice. Now it make sense.